src/main/java/com/mraof/minestuck/client/gui/playerStats/EcheladderScreen.java

```diff
@@ -179,21 +186,62 @@ private void drawLadder(PoseStack poseStack, int currentRung, boolean showLastRu
 				textColor = rand.nextInt(0xFFFFFF);
 				if(textColors.length > rung)
 					textColor = textColors[rung];
-				fill(poseStack, xOffset + 90, y, xOffset + 236, y + 12, backgrounds.length > rung ? backgrounds[rung] : (textColor^0xFFFFFFFF));
+				fill(poseStack, xOffset + 90, y, xOffset + 236, y + 12, backgrounds.length > rung ? backgrounds[rung] : (textColor ^ 0xFFFFFFFF));
 			} else if(rung == currentRung + 1 && animationCycle == 0)
 			{
 				int bg = ~rand.nextInt(0xFFFFFF);
 				if(backgrounds.length > rung)
 					bg = backgrounds[rung];
 				else if(textColors.length > rung)
 					bg = ~textColors[rung];
-				fill(poseStack, xOffset + 90, y + 10, xOffset + 90 + (int)(146* ClientPlayerData.getRungProgress()), y + 12, bg);
+				fill(poseStack, xOffset + 90, y + 10, xOffset + 90 + (int) (146 * ClientPlayerData.getRungProgress()), y + 12, bg);
 			} else rand.nextInt(0xFFFFFF);
 			
-			String s = I18n.exists("echeladder.rung."+rung) ? I18n.get("echeladder.rung."+rung) : "Rung "+(rung+1);
-			mc.font.draw(poseStack, s, xOffset+ladderXOffset - mc.font.width(s) / 2, y + 2, textColor);
+			
+			String s = null;
+			if(playerTitle != null)
+			{
+				String playerBoots = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(0));
+				String playerLeggings = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(1));
+				String playerChestplate = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(2));
+				String playerHat = String.valueOf((Minecraft.getInstance().player).getInventory().armor.get(3));
+				String playerClass = (playerTitle.getHeroClass().toString());
+				String playerAspect = (playerTitle.getHeroAspect().toString());
+				String playerColor = String.valueOf(ClientPlayerData.getPlayerColor());
+				String playerBoon = String.valueOf(ClientPlayerData.getBoondollars());
+				String playerModus = String.valueOf(ClientPlayerData.getModus());
+				String consortRep = String.valueOf(ClientPlayerData.getConsortReputation());
+				String isWearingArmor = playerChestplate + playerBoots + playerHat + playerLeggings;
+				
+				if(I18n.exists("echeladder.rung." + rung + "." + playerAspect + "." + playerClass))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerAspect + "." + playerClass);
+				}
+				else if(I18n.exists("echeladder.rung." + rung + "." + playerColor))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerColor);
+				}
+				
+				else if(I18n.exists("echeladder.rung." + rung + "." + playerAspect))
+				{
+					s = I18n.get("echeladder.rung." + rung + "." + playerAspect);
+				}
+			}
+				if(s == null)
+				{
+					if(I18n.exists("echeladder.rung." + rung))
+					{
+						s = I18n.get("echeladder.rung." + rung);
+					}
+					else
+					{
+						s = "Rung " + (rung + 1);
+					}
+				}
+				mc.font.draw(poseStack, s, xOffset + ladderXOffset -
+						mc.font.width(s) / 2, y + 2, textColor);
```

Comment:
It's probably better to make this one line.

Reply:
For the second part of your comment, see my comment on EcheladderScreen